anchor/processor/src/lib.rs

```
@@ -0,0 +1,145 @@
+use serde::{Deserialize, Serialize};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use task_executor::TaskExecutor;
+use tokio::select;
+use tokio::sync::mpsc::error::TrySendError;
+use tokio::sync::{mpsc, Semaphore};
+use tracing::{error, warn};
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Config {
+    pub max_workers: usize,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            max_workers: num_cpus::get(),
+        }
+    }
+}
+
+pub struct Sender {
+    name: &'static str,
+    tx: mpsc::Sender<WorkItem>,
+}
+
+impl Sender {
+    fn new(name: &'static str, tx: mpsc::Sender<WorkItem>) -> Self {
+        Self { name, tx }
+    }
+
+    pub fn send_async(&mut self, future: AsyncFn) {
+        self.send_work_item(WorkItem::new_async(self.name, future));
+    }
+
+    pub fn send_blocking(&mut self, func: BlockingFn) {
+        self.send_work_item(WorkItem::new_blocking(self.name, func));
+    }
+
+    pub fn send_work_item(&mut self, item: WorkItem) {
```

At some point, we might need to return an error here to handle the case of the queue being full

---

Good idea, done